File: THCS/2022_2023/ThanhHoa_2223/Bai1/TAMGIAC.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main() {
    // Đọc dữ liệu từ file
    ifstream infile("TAMGIAC.INP");
    ofstream outfile("TAMGIAC.OUT");

    int T;
    infile >> T; // Số bộ test

    for (int i = 0; i < T; ++i) {
        int a, b, c;
        infile >> a >> b >> c; // Đọc ba số góc

        // Kiểm tra điều kiện của tam giác
        if (a + b + c != 180 || a <= 0 || b <= 0 || c <= 0) {
            outfile << 0 << endl; // Không phải là số đo của tam giác
        } else {
            // Xác định loại tam giác
            if (a < 90 && b < 90 && c < 90) {
                outfile << "NHON" << endl; // Tam giác nhọn
            } else if (a == 90 || b == 90 || c == 90) {
                outfile << "VUONG" << endl; // Tam giác vuông
            } else {
                outfile << "TU" << endl; // Tam giác tù
            }
        }
    }

    infile.close();
    outfile.close();

    return 0;
}

File: THCS/2022_2023/ThanhHoa_2223/Bai1/TAMGIAC.py
def check_triangle_type(a, b, c):
    # Kiểm tra tổng ba góc có bằng 180 không và các góc có lớn hơn 0 không
    if a + b + c != 180 or a <= 0 or b <= 0 or c <= 0:
        return 0  # Không phải là số đo của tam giác
    else:
        # Xác định loại tam giác
        if a < 90 and b < 90 and c < 90:
            return "NHON"  # Tam giác nhọn
        elif a == 90 or b == 90 or c == 90:
            return "VUONG"  # Tam giác vuông
        else:
            return "TU"  # Tam giác tù

def main():
    # Đọc dữ liệu từ file
    with open("TAMGIAC.INP", "r") as infile:
        T = int(infile.readline().strip())  # Số bộ test

        results = []
        for _ in range(T):
            a, b, c = map(int, infile.readline().strip().split())  # Đọc ba số góc
            result = check_triangle_type(a, b, c)
            results.append(result)

    # Ghi kết quả vào file
    with open("TAMGIAC.OUT", "w") as outfile:
        for result in results:
            outfile.write(str(result) + "\n")

if __name__ == "__main__":
    main()
